src/MongoDB.Driver.Core/ServerErrorCode.cs

```diff
@@ -21,6 +21,7 @@ internal enum ServerErrorCode
         // see: https://github.com/mongodb/mongo/blob/master/src/mongo/base/error_codes.yml
         CappedPositionLost = 136,
         CursorKilled = 237,
+        CursorNotFound = 43,
```

Comment:
we have MongoCursorNotFoundException exception

Reply:
Looks like we're not correctly throwing a MongoCursorNotFoundException from everywhere it should be thrown.

Reply:
yes. And my point whether we should add it into exceptionMapper logic or no. I mentioned it because previously we had this exception in the retryable logic: https://github.com/DmitryLukyanov/mongo-csharp-driver/commit/ea4d8d3e5ff7b7cb6d411ce36096b53c3e358373#diff-f3cf36bc2821907b3003ced21fae1593L49. Now sure how it worked.
But in any case it sounds like we need to consider avoid two different code paths for the same `cursorNotFound` logic

Reply:
Added mapping, but the error code wasn't removed.
Not done.